pkg/services/ngalert/state/clean.go

```diff
@@ -0,0 +1,64 @@
+package state
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	ngModels "github.com/grafana/grafana/pkg/services/ngalert/models"
+)
+
+// Clean removes superfluous and orphan instances from the database.
+// It does not affect the manager's loaded cache. The manager will not load orphan states to begin with.
+// If orphan states are present in the cache, they will be left there.
+func (st *Manager) Clean(ctx context.Context, rules RuleReader) error {
+	return Clean(ctx, st.instanceStore, rules, st.log)
+}
+
+// Clean removes orphan instances from `instances`, based on the rules given by `rules`.
+func Clean(ctx context.Context, instances InstanceStore, rules RuleReader, logger log.Logger) error {
```

Comment:
nit: I think we can make this package-private since its called from `manager.Clean`?

```suggestion
func clean(ctx context.Context, instances InstanceStore, rules RuleReader, logger log.Logger) error {
```